src/support.rs

```diff
@@ -173,10 +176,6 @@ fn assert_unique_ptr_layout_compatible<U, T>() {
   // as a raw C pointer.
   assert_eq!(size_of::<U>(), size_of::<*mut T>());
   assert_eq!(align_of::<U>(), align_of::<*mut T>());
-
-  // Assert that `T` (probably) implements `Drop`. If it doesn't, a regular
-  // reference should be used instead of UniquePtr/UniqueRef.
-  assert!(needs_drop::<T>());
```

Comment:
Removed because I don't think it makes sense to enforce this, the comment is wrong in that a regular reference is not a safe replacement since it would carry aliasing invariants.

Also in general I prefer only using raw pointers to refer to C++ values, and implementing drop forces you to take a mutable ref